cupyx/scipy/sparse/compressed.py

```diff
@@ -480,57 +483,79 @@ def __sub__(self, other):
     def __rsub__(self, other):
         return self._add(other, True, False)
 
-    def __getitem__(self, slices):
-        if isinstance(slices, tuple):
-            slices = list(slices)
-        elif isinstance(slices, list):
-            slices = list(slices)
-            if all([isinstance(s, int) for s in slices]):
-                slices = [slices]
-        else:
-            slices = [slices]
-
-        ellipsis = -1
-        n_ellipsis = 0
-        for i, s in enumerate(slices):
-            if s is None:
-                raise IndexError('newaxis is not supported')
-            elif s is Ellipsis:
-                ellipsis = i
-                n_ellipsis += 1
-        if n_ellipsis > 0:
-            ellipsis_size = self.ndim - (len(slices) - 1)
-            slices[ellipsis:ellipsis + 1] = [slice(None)] * ellipsis_size
-
-        if len(slices) == 2:
-            row, col = slices
-        elif len(slices) == 1:
-            row, col = slices[0], slice(None)
-        else:
-            raise IndexError('invalid number of indices')
+    def _get_intXint(self, row, col):
+        M, N = self._swap(*self.shape)
+        major, minor = self._swap(row, col)
+
+        indptr, indices, data = _index._get_csr_submatrix(
+            self.indptr, self.indices, self.data,
+            major, major + 1, minor, minor + 1)
+        return data.sum(dtype=self.dtype)
+
+    def _get_sliceXslice(self, row, col):
 
         major, minor = self._swap(row, col)
-        major_size, minor_size = self._swap(*self._shape)
-        if numpy.isscalar(major):
-            i = int(major)
-            if i < 0:
-                i += major_size
-            if not (0 <= i < major_size):
-                raise IndexError('index out of bounds')
-            if numpy.isscalar(minor):
-                j = int(minor)
-                if j < 0:
-                    j += minor_size
-                if not (0 <= j < minor_size):
-                    raise IndexError('index out of bounds')
-                return self._get_single(i, j)
-            elif minor == slice(None):
-                return self._get_major_slice(slice(i, i + 1))
-        elif isinstance(major, slice):
-            if minor == slice(None):
-                return self._get_major_slice(major)
-
-        raise ValueError('unsupported indexing')
+        if major.step in {1, None} and minor.step in {1, None}:
```

Comment:
Please use tuples instead of sets.
```suggestion
        if major.step in (1, None) and minor.step in (1, None):
```